pkg/utils/managedresources/managedresources.go

```diff
@@ -263,14 +270,14 @@ func WaitUntilListDeleted(ctx context.Context, client client.Client, mrList *res
 }
 
 // WaitUntilDeleted waits until the given managed resource is deleted.
-func WaitUntilDeleted(ctx context.Context, client client.Client, namespace, name string) error {
+func WaitUntilDeleted(ctx context.Context, clientPar client.Client, namespace, name string) error {
```

Comment:
The doc string of the `WaitUntilDeleted` func has to be adapted that it now also waits the MR Secrets are deleted.